### ISSUE TITLE
Can't build in Xcode 11 after upgrade to Expo SDK 35.0.0 from 33.0.0

### ISSUE THREAD

**Issue by mfkrause** [expo/expo#5728]
## 🐛 Bug Report
I'm trying to upgrade from Expo SDK 35.0.0 (from previously 33.0.0) since it's said to have fixed the `unknown argument type '__attribute__' issue` in Xcode 11. However, I can't manage to build my app now. I've updated Expo SDK, Expo CLI, all my node_modules and Pods; relinked every native library as well (I'm detached from the managed Expo workflow).

### Environment
```
Expo CLI 3.1.0 environment info:
    System:
      OS: macOS 10.14.6
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 10.15.1 - /usr/local/bin/node
      npm: 6.4.1 - /usr/local/bin/npm
    IDEs:
      Android Studio: 3.5 AI-191.8026.42.35.5791312
      Xcode: 11.0/11A420a - /usr/bin/xcodebuild
    npmPackages:
      expo: ^35.0.0 => 35.0.0
      react: 16.8.3 => 16.8.3
      react-native: https://github.com/expo/react-native/archive/sdk-35.0.0.tar.gz => 0.59.8
      react-navigation: ^3.11.1 => 3.11.1
    npmGlobalPackages:
      expo-cli: 3.1.0
```

### Steps to Reproduce

1. Have a project running Expo SDK 33.0.0, ejected and with native modules linked.
2. Upgrade Expo SDK to 35.0.0.
3. Delete node_modules and reinstall the modules.
4. Unlink every package through `react-native unlink`
4. Update the CocoaPods repo and upgrade the Pods.
5. Relink everything (some packages manually).
6. Run the application from Xcode 11.

### Expected Behavior
App should build.

### Actual Behavior
Apple Mach-O Linker Error
```
ld: warning: directory not found for option '-L/Users/maximiliankrause/Library/Developer/Xcode/DerivedData/teledoctor24-dmzpmlxqrvleredkktkzbzcmwtdi/Build/Products/Debug-iphonesimulator/EXAV'
ld: warning: directory not found for option '-L/Users/maximiliankrause/Library/Developer/Xcode/DerivedData/teledoctor24-dmzpmlxqrvleredkktkzbzcmwtdi/Build/Products/Debug-iphonesimulator/EXAdsAdMob'
ld: warning: directory not found for option '-L/Users/maximiliankrause/Library/Developer/Xcode/DerivedData/teledoctor24-dmzpmlxqrvleredkktkzbzcmwtdi/Build/Products/Debug-iphonesimulator/EXAdsFacebook'
ld: warning: directory not found for option '-L/Users/maximiliankrause/Library/Developer/Xcode/DerivedData/teledoctor24-dmzpmlxqrvleredkktkzbzcmwtdi/Build/Products/Debug-iphonesimulator/EXAmplitude'
ld: warning: directory not found for option '-L/Users/maximiliankrause/Library/Developer/Xcode/DerivedData/teledoctor24-dmzpmlxqrvleredkktkzbzcmwtdi/Build/Products/Debug-iphonesimulator/EXAppAuth'
ld: warning: directory not found for option '-L/Users/maximiliankrause/Library/Developer/Xcode/DerivedData/teledoctor24-dmzpmlxqrvleredkktkzbzcmwtdi/Build/Products/Debug-iphonesimulator/EXBackgroundFetch'
ld: warning: directory not found for option '-L/Users/maximiliankrause/Library/Developer/Xcode/DerivedData/teledoctor24-dmzpmlxqrvleredkktkzbzcmwtdi/Build/Products/Debug-iphonesimulator/EXBarCodeScanner'
ld: warning: directory not found for option '-L/Users/maximiliankrause/Library/Developer/Xcode/DerivedData/teledoctor24-dmzpmlxqrvleredkktkzbzcmwtdi/Build/Products/Debug-iphonesimulator/EXBlur'
ld: warning: directory not found for option '-L/Users/maximiliankrause/Library/Developer/Xcode/DerivedData/teledoctor24-dmzpmlxqrvleredkktkzbzcmwtdi/Build/Products/Debug-iphonesimulator/EXBrightness'
ld: warning: directory not found for option '-L/Users/maximiliankrause/Library/Developer/Xcode/DerivedData/teledoctor24-dmzpmlxqrvleredkktkzbzcmwtdi/Build/Products/Debug-iphonesimulator/EXCalendar'
ld: warning: directory not found for option '-L/Users/maximiliankrause/Library/Developer/Xcode/DerivedData/teledoctor24-dmzpmlxqrvleredkktkzbzcmwtdi/Build/Products/Debug-iphonesimulator/EXCamera'
ld: warning: directory not found for option '-L/Users/maximiliankrause/Library/Developer/Xcode/DerivedData/teledoctor24-dmzpmlxqrvleredkktkzbzcmwtdi/Build/Products/Debug-iphonesimulator/EXContacts'
ld: warning: directory not found for option '-L/Users/maximiliankrause/Library/Developer/Xcode/DerivedData/teledoctor24-dmzpmlxqrvleredkktkzbzcmwtdi/Build/Products/Debug-iphonesimulator/EXCrypto'
ld: warning: directory not found for option '-L/Users/maximiliankrause/Library/Developer/Xcode/DerivedData/teledoctor24-dmzpmlxqrvleredkktkzbzcmwtdi/Build/Products/Debug-iphonesimulator/EXDocumentPicker'
ld: warning: directory not found for option '-L/Users/maximiliankrause/Library/Developer/Xcode/DerivedData/teledoctor24-dmzpmlxqrvleredkktkzbzcmwtdi/Build/Products/Debug-iphonesimulator/EXFacebook'
ld: warning: directory not found for option '-L/Users/maximiliankrause/Library/Developer/Xcode/DerivedData/teledoctor24-dmzpmlxqrvleredkktkzbzcmwtdi/Build/Products/Debug-iphonesimulator/EXGL'
ld: warning: directory not found for option '-L/Users/maximiliankrause/Library/Developer/Xcode/DerivedData/teledoctor24-dmzpmlxqrvleredkktkzbzcmwtdi/Build/Products/Debug-iphonesimulator/EXGL-CPP'
ld: warning: directory not found for option '-L/Users/maximiliankrause/Library/Developer/Xcode/DerivedData/teledoctor24-dmzpmlxqrvleredkktkzbzcmwtdi/Build/Products/Debug-iphonesimulator/EXGoogleSignIn'
ld: warning: directory not found for option '-L/Users/maximiliankrause/Library/Developer/Xcode/DerivedData/teledoctor24-dmzpmlxqrvleredkktkzbzcmwtdi/Build/Products/Debug-iphonesimulator/EXHaptics'
ld: warning: directory not found for option '-L/Users/maximiliankrause/Library/Developer/Xcode/DerivedData/teledoctor24-dmzpmlxqrvleredkktkzbzcmwtdi/Build/Products/Debug-iphonesimulator/EXImageManipulator'
ld: warning: directory not found for option '-L/Users/maximiliankrause/Library/Developer/Xcode/DerivedData/teledoctor24-dmzpmlxqrvleredkktkzbzcmwtdi/Build/Products/Debug-iphonesimulator/EXLocalAuthentication'
ld: warning: directory not found for option '-L/Users/maximiliankrause/Library/Developer/Xcode/DerivedData/teledoctor24-dmzpmlxqrvleredkktkzbzcmwtdi/Build/Products/Debug-iphonesimulator/EXLocalization'
ld: warning: directory not found for option '-L/Users/maximiliankrause/Library/Developer/Xcode/DerivedData/teledoctor24-dmzpmlxqrvleredkktkzbzcmwtdi/Build/Products/Debug-iphonesimulator/EXMailComposer'
ld: warning: directory not found for option '-L/Users/maximiliankrause/Library/Developer/Xcode/DerivedData/teledoctor24-dmzpmlxqrvleredkktkzbzcmwtdi/Build/Products/Debug-iphonesimulator/EXMediaLibrary'
ld: warning: directory not found for option '-L/Users/maximiliankrause/Library/Developer/Xcode/DerivedData/teledoctor24-dmzpmlxqrvleredkktkzbzcmwtdi/Build/Products/Debug-iphonesimulator/EXPrint'
ld: warning: directory not found for option '-L/Users/maximiliankrause/Library/Developer/Xcode/DerivedData/teledoctor24-dmzpmlxqrvleredkktkzbzcmwtdi/Build/Products/Debug-iphonesimulator/EXRandom'
ld: warning: directory not found for option '-L/Users/maximiliankrause/Library/Developer/Xcode/DerivedData/teledoctor24-dmzpmlxqrvleredkktkzbzcmwtdi/Build/Products/Debug-iphonesimulator/EXSMS'
ld: warning: directory not found for option '-L/Users/maximiliankrause/Library/Developer/Xcode/DerivedData/teledoctor24-dmzpmlxqrvleredkktkzbzcmwtdi/Build/Products/Debug-iphonesimulator/EXSegment'
ld: warning: directory not found for option '-L/Users/maximiliankrause/Library/Developer/Xcode/DerivedData/teledoctor24-dmzpmlxqrvleredkktkzbzcmwtdi/Build/Products/Debug-iphonesimulator/EXSensors'
ld: warning: directory not found for option '-L/Users/maximiliankrause/Library/Developer/Xcode/DerivedData/teledoctor24-dmzpmlxqrvleredkktkzbzcmwtdi/Build/Products/Debug-iphonesimulator/EXSharing'
ld: warning: directory not found for option '-L/Users/maximiliankrause/Library/Developer/Xcode/DerivedData/teledoctor24-dmzpmlxqrvleredkktkzbzcmwtdi/Build/Products/Debug-iphonesimulator/EXSpeech'
ld: warning: directory not found for option '-L/Users/maximiliankrause/Library/Developer/Xcode/DerivedData/teledoctor24-dmzpmlxqrvleredkktkzbzcmwtdi/Build/Products/Debug-iphonesimulator/EXTaskManager'
ld: library not found for -lEXAV
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

/**edit**: Might be related. I get this warning during pods installation:
```
Found some duplicated unimodule packages. Installed the ones with the highest version number.
Make sure following dependencies of your project are resolving to one specific version:
 expo-image-picker, unimodules-sensors-interface, expo-file-system, unimodules-image-loader-interface, unimodules-face-detector-interface, unimodules-constants-interface, unimodules-font-interface, expo-location, unimodules-file-system-interface, expo-constants, unimodules-camera-interface, expo-permissions, unimodules-task-manager-interface, unimodules-permissions-interface, unimodules-react-native-adapter, unimodules-core, expo-app-loader-provider, unimodules-barcode-scanner-interface
```

I've noticed I forgot to update the ExpoKit SDK versions in ios/<project/>Supporting/EXSDKVersions.plist as well as change the ExpoKit release version in the Podfile (I now changed it to ios/2.13.0). Did both of this now and reinstalled the pods. Still the same error though.
After having done so, this warning now got added to the list:
`ld: warning: directory not found for option '-F/Users/maximiliankrause/teledoctor24-app/ios/Pods/Google-Mobile-Ads-SDK/Frameworks/frameworks'`

During installation of the new ios/2.13.0 ExpoKit, CocoaPods removed this package. So, I suppose all this comes from removed ExpoKit libraries still trying to be linked - somewhere. I have no idea where though.

**Comment by Szymon20000** [comment]
I had no problems with the upgrade from 33 to 35.  Maybe my steps can help you.

1.  `rm -rf node_modules`
2.  update `expo` in `package.json` from `33.0.0` to `35.0.0`
3.  update `react-native-unimodules` to `0.5.2`
4.  update `react-native` replace `.../sdk-33.0.0.tar.gz`with `.../sdk-35.0.0.tar.gz`
5.  remove `expokit` dependency from package.json
6. yarn
7.  replace sdkVersion in `app.json`
8. update ExpoKit tag 
``` 
  :tag => "ios/2.13.0",
```
9.  pod update
10. pod install
11. replace values in `os/your-project/Supporting/EXSDKVersions.plist`

**Comment by Flaque** [comment]
@Szymon20000 Why did you remove `expokit` dependency from package.json?

**Comment by Szymon20000** [comment]
You are right! It must stay there. It's not used on iOS and that's why I removed it. But we still use it on Android.

**Comment by jvgaeta** [comment]
I'm having the same issue. It seems to be an issue with the linker flags. What are your other linker flags in the xcode project set to @Szymon20000 ? Thanks!

**Comment by jvgaeta** [comment]
Found a solution. Take a look at your other linker flags and make sure it's not referencing things not installed via pod install. If you remove these references, it will fix your build. https://github.com/expo/expo/issues/5760

**Comment by virtual-marat** [comment]
Hi @jvgaeta! Where is search other linker flags? I create new pet project and use `npm run android` for run project

**Comment by jvgaeta** [comment]
@virtual-marat This is for ios. You navigate to your project -> Build Settings. Then there's a row called 'Other Linker Flags'. There's a picture in this thread of what it looks like: https://github.com/expo/expo/issues/5760

**Comment by mfkrause** [comment]
@Szymon20000 @jvgaeta Thank you two. Reverted all my changes a while ago and now tried again to update to SDK 35 since I officially couldn't build with Xcode 10 anymore, even using workarounds.
I first did the steps @Szymon20000 wrote. I then ran into linker issues again, however I then actually just removed them all as @jvgaeta said - had to remove Bolts, the EX* packages which were not installed by CocoaPods earlier, and a Google package. After that, the app built! Ran into several package and deprecation issues afterwards which I'm now fixing one by one. However, everything seems to work fine so far.
